drivers/overwrite_queue_test.go

```diff
@@ -0,0 +1,70 @@
+package drivers
+
+import (
+	"errors"
+	"testing"
+	"time"
+)
+
+func TestOverwriteQueueShouldCallSave(t *testing.T) {
+	timeout := 500 * time.Millisecond
+	mos := &MockOSSession{}
+	oq := NewOverwriteQueue(mos, "f1", "save", 2, timeout, 15*time.Second)
+
+	data1 := []byte("data01")
+
+	var meta map[string]string
+	mos.On("SaveData", "f1", data1, meta, timeout).Return("not used", nil).Once()
+
+	oq.Save(data1)
+	oq.waitForQueueToClear(5 * time.Second)
+	mos.AssertExpectations(t)
+	oq.StopAfter(0)
+	time.Sleep(10 * time.Millisecond)
+}
+
+func TestOverwriteQueueShouldRetry(t *testing.T) {
+	timeout := 500 * time.Millisecond
+	mos := &MockOSSession{}
+	oq := NewOverwriteQueue(mos, "f1", "retry", 2, timeout, 15*time.Second)
+
+	data1 := []byte("data01")
+
+	var meta map[string]string
+	mos.On("SaveData", "f1", data1, meta, timeout).Return("not used", errors.New("no1")).Once()
+	timeout = time.Duration(float64(timeout) * timeoutMultiplier)
+	mos.On("SaveData", "f1", data1, meta, timeout).Return("not used", nil).Once()
+
+	oq.Save(data1)
+	oq.waitForQueueToClear(5 * time.Second)
+	mos.AssertExpectations(t)
+	oq.StopAfter(0)
+	time.Sleep(10 * time.Millisecond)
+}
+
+func TestOverwriteQueueShouldUseLastValue(t *testing.T) {
+	timeout := 300 * time.Millisecond
+	mos := &MockOSSession{}
+	mos.init()
+	oq := NewOverwriteQueue(mos, "f1", "use last", 2, timeout, 15*time.Second)
+
+	dataw1 := []byte("dataw01")
+	data2 := []byte("data02")
+	data3 := []byte("data03")
+
+	var meta map[string]string
+	mos.On("SaveData", "f1", dataw1, meta, timeout).Return("not used", nil).Once()
+	mos.On("SaveData", "f1", data3, meta, timeout).Return("not used", nil).Once()
+
+	mos.waitForCh = true
+	oq.Save(dataw1)
+	<-mos.back
+	oq.Save(data2)
+	oq.Save(data3)
+	mos.waitCh <- nil
+
+	oq.waitForQueueToClear(5 * time.Second)
+	mos.AssertExpectations(t)
```

Comment:
for completeness, we could also add assertion for SaveData not being called for "data02" here using https://pkg.go.dev/github.com/stretchr/testify/mock#Mock.AssertNotCalled

currently it may get called along with dataw01, and data03 but we'll miss that in the test.

Reply:
added